src/main/java/seedu/address/model/asset/Asset.java

```diff
@@ -11,26 +12,25 @@
  */
 public class Asset {
 
-    public static final String MESSAGE_CONSTRAINTS = "Asset names can take any values, and it should not be blank";
-    private static final String VALIDATION_REGEX = "\\S.*";
+    public static final String MESSAGE_CONSTRAINTS = "Assets must be entered in the format NAME[#ID]";
+    private static final String VALIDATION_REGEX = "\\s*([^#\\s]+\\s*)+(#(\\s*[^#\\s]+\\s*)+)?";
 
     private final String assetName;
+    private final String assetId;
 
-    /**
-     * Constructs a {@code Asset}.
-     *
-     * @param assetName A valid asset name.
-     */
-    public Asset(String assetName) {
-        requireNonNull(assetName);
-        assetName = assetName.trim();
-        checkArgument(isValid(assetName), MESSAGE_CONSTRAINTS);
+    private Asset(String assetName, String assetId) {
+        requireAllNonNull(assetName, assetId);
         this.assetName = assetName;
+        this.assetId = assetId;
     }
 
     @JsonValue
     public String get() {
-        return assetName;
+        StringBuilder res = new StringBuilder(assetName);
+        if (!assetId.isEmpty()) {
+            res.append("#").append(assetId);
+        }
+        return res.toString();
```

Comment:
Is the use of `StringBuilder` necessary here?

Reply:
Changed such that it isn't here anymore.